src/main/java/org/bookkepervisualmanager/api/resources/LedgersResource.java

```diff
@@ -0,0 +1,137 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedMap;
+import javax.ws.rs.Produces;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+import org.apache.bookkeeper.client.api.LedgerMetadata;
+
+@Path("ledger")
+public class LedgersResource extends AbstractBookkeeperResource {
+
+    @GET
+    @Path("all")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<LedgerBean> getLedgers() throws Exception {
+        final List<LedgerBean> ledgers = new ArrayList<>();
+
+        Map<Long, LedgerMetadata> allLedgers = getBookkeeperManger().getAllLedgers();
+        for (Entry<Long, LedgerMetadata> currentLedger : allLedgers.entrySet()) {
+            LedgerBean b = new LedgerBean();
+
+            long lid = currentLedger.getKey();
+            b.setId(lid);
+
+            LedgerMetadata ledgerMetadata = currentLedger.getValue();
+            Map<String, byte[]> customMetadata = ledgerMetadata.getCustomMetadata();
+            for (Entry<String, byte[]> currentCustomMetadata : customMetadata.entrySet()) {
+                b.setMetadataValue(currentCustomMetadata.getKey(), currentCustomMetadata.getValue());
+            }
+            ledgers.add(b);
+        };
+
+        return ledgers;
+    }
+
+    @GET
+    @Path("metadata/{ledgerId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public LedgerBean getLedgerMetadata(@PathParam("ledgerId") long ledgerId) throws Exception {
+
+        LedgerMetadata ledgerMetadata = getBookkeeperManger().getLedgerMetadata(ledgerId);
+        LedgerBean b = new LedgerBean();
+        b.setId(ledgerId);
+
+        Map<String, byte[]> customMetadata = ledgerMetadata.getCustomMetadata();
+        for (Entry<String, byte[]> currentCustomMetadata : customMetadata.entrySet()) {
+            b.setMetadataValue(currentCustomMetadata.getKey(), currentCustomMetadata.getValue());
+        }
+
+        return b;
+    }
+
+    @GET
+    @Path("bookie/{bookieId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<LedgerBean> getLedgersForBookie(@PathParam("bookieId") String bookieId) throws Exception {
+        final List<LedgerBean> ledgers = new ArrayList<>();
+
+        SortedMap<Long, LedgerMetadata> forBookie = getBookkeeperManger().getLedgersForBookie(bookieId);
+        for (Entry<Long, LedgerMetadata> currentLedger : forBookie.entrySet()) {
+            LedgerBean b = new LedgerBean();
+
+            long lid = currentLedger.getKey();
+            b.setId(lid);
+
+            LedgerMetadata ledgerMetadata = currentLedger.getValue();
+            Map<String, byte[]> customMetadata = ledgerMetadata.getCustomMetadata();
+            for (Entry<String, byte[]> currentCustomMetadata : customMetadata.entrySet()) {
+                b.setMetadataValue(currentCustomMetadata.getKey(), currentCustomMetadata.getValue());
```

Comment:
convert byte[] -> string(utf-8)

create an utility method to map LergerMetadata -> LedgerBean